xbmc/GUIInfoManager.cpp

```diff
@@ -2654,7 +2662,8 @@ const infomap musicplayer[] =    {{ "title",            MUSICPLAYER_TITLE },
                                   { "originaldate",     MUSICPLAYER_ORIGINALDATE },
                                   { "bpm",              MUSICPLAYER_BPM },
                                   { "ismultidisc",      MUSICPLAYER_ISMULTIDISC },
-                                  { "totaldiscs",       MUSICPLAYER_TOTALDISCS }
+                                  { "totaldiscs",       MUSICPLAYER_TOTALDISCS },
+                                  { "station",          MUSICPLAYER_STATIONNAME }
```

Comment:
We already have "MusicPlayer.ChannelName", which we currently use for PVR Radio channel names. Would fit nicely also for the shoucast station name, imo. No new label needed.

Reply:
I'm aware of that but, for instance, iptvsimple can set its station name from the playlist where it can be user-defined, but if I re-used that label, it would be overwritten. I'd rather not re-use a label that has the potential to already have been set by something else.

Reply:
Fair enough. May I ask how that will work to set a station name in the iptvsimple addon which then will be available in kodi?

Reply:
Put something like this in an m3u8 file and point iptvsimple to it.
```
#EXTINF:0 radio="true", Planet Rock
https://stream-mz.planetradio.co.uk/planetrock.mp3
#EXTINF:0 radio="true", Virgin Radio UK
http://radio.virginradio.co.uk/stream
```
iptvsimple will set the station name as specified there via `MusicPlayer.ChannelName`. That gets displayed in the PVR overlay.

Reply:
> iptvsimple will set the station name as specified there via MusicPlayer.ChannelName. That gets displayed in the PVR overlay.

iptvsimple cannot directly set a guiinfo label. Please explain how exactly iptvsimple will set the information that eventually gets used for the new guiinfo label MusicPlayer.StationName

Reply:
It doesn't, and it isn't meant to.  The PVR side has it's own set of labels that iptvsimple uses (channelNumber and channelName).  Whatever I set in the `#EXTINF:0` line as the station name gets put into `MusicPlayer.ChannelName` and is displayed by a completely different overlay in Estuary. So in the above example, it would show `Virgin Radio UK`. If I re-used that label for the shoutcast stream it would show whatever is set in the header for that station, which would overwrite the user specified value and display `VIRGIN_128_UK`  The PVR stuff is completely untouched by anything in this PR, which is intended, other than the fact that the art can be displayed (but that's a skin thing anyway).

Reply:
I know that PVR stuff is not touched and this is not my concern. I just want to understand how iptvsimple as a pvr addon is able to transport the non-pvr data to kodi. is it just http headers, because kodi does HTTP GET to serve the shoutcast stream it gets from iptvsimple. Sorry for asking such dumb questions - I never before dealt with that.

Reply:
Hey, it's not a dumb question!  TBH I don't know exactly how it does it, I just know that it does and that in my original PR, concerns were raised over how my original implementation could/would affect iptvsimple.  @phunkyfish or @emveepee are better able to answer that.